From 4edaed6d6b46e3d55a51d8cd0291c4276d8a9634 Mon Sep 17 00:00:00 2001
From: Joerg Sonnenberger <joerg@bec.de>
Date: Sun, 1 May 2016 22:03:23 +0200
Subject: [PATCH] joyent/mdata-client#7 Rename freebsd.c to bsd.c. Add NetBSD
 support. Reviewed by: Robert Mustacchi <rm@joyent.com> Approved by: Robert
 Mustacchi <rm@joyent.com>

---
 Makefile                  | 14 +++++++++++++-
 plat/{freebsd.c => bsd.c} |  4 ++++
 2 files changed, 17 insertions(+), 1 deletion(-)
 rename plat/{freebsd.c => bsd.c} (97%)

diff --git a/Makefile b/Makefile
index 374d16d..b464961 100644
--- a/Makefile
+++ b/Makefile
@@ -68,7 +68,19 @@ CTFMERGE = /usr/bin/true
 CTFCONVERT = /usr/bin/true
 
 CFLAGS += -Wno-typedef-redefinition
-CFILES += plat/freebsd.c plat/unix_common.c
+CFILES += plat/bsd.c plat/unix_common.c
+HDRS += plat/unix_common.h
+PLATFORM_OK = true
+MANSECT = 1
+endif
+
+ifeq ($(UNAME_S),NetBSD)
+CC != if [ -x /usr/bin/clang ]; then echo /usr/bin/clang; else echo /usr/bin/gcc; fi
+
+CTFMERGE != if [ -x /usr/bin/ctfmerge ]; then echo /usr/bin/ctfmerge; else echo /usr/bin/true; fi
+CTFCONVERT != if [ -x /usr/bin/ctfconvert ]; then echo /usr/bin/ctfconvert; else echo /usr/bin/true; fi
+
+CFILES += plat/bsd.c plat/unix_common.c
 HDRS += plat/unix_common.h
 PLATFORM_OK = true
 MANSECT = 1
diff --git a/plat/freebsd.c b/plat/bsd.c
similarity index 97%
rename from plat/freebsd.c
rename to plat/bsd.c
index 7f30386..dfb3aa3 100644
--- a/plat/freebsd.c
+++ b/plat/bsd.c
@@ -25,7 +25,11 @@
 #include "dynstr.h"
 #include "plat/unix_common.h"
 
+#ifdef __NetBSD__
+#define	SERIAL_DEVICE	"/dev/tty01"
+#else
 #define	SERIAL_DEVICE	"/dev/cuau1"
+#endif
 
 typedef struct mdata_plat {
 	int mpl_kq;
-- 
2.21.0

